User Saved Images View

edimarvalentin opened the pull request:
# Feature Description
This feature implements the ability for users to view their own designs from the gallery in a paginated fashion. When a design is deleted, the UI updates instantly by removing the deleted design from the gallery without needing a full refresh from the server. A confirmation modal is displayed before deletion to ensure intentional user actions. Users can also queue or edit their images using this view as a starting point. 
# Solution Description
Upon clicking "Delete," a confirmation modal appears with the selected design's title. If the user confirms, the frontend sends a DELETE request to the backend API with the design ID in a JSON payload. After a successful deletion, the deleted design is immediately removed from the designs array in the frontend state. If no designs remain, the app navigates to the previous page or clears the selection. 
# Additional Changes
- Updated the header and footer to improve layout consistency across all pages.
- The footer now stays fixed at the bottom of the screen and no longer overlaps or hides content.
- The background behind modals is now properly blurred and darkened for a more professional and focused appearance.

# Preview
![Screenshot 2025-04-30 at 16-42-31 Vite React](https://github.com/user-attachments/assets/3583b41a-f9dc-47fd-b43a-fd795c4be245)

# Copilot Summary:
This pull request introduces several key updates across the backend and frontend, focusing on new API endpoints, backend logic improvements, and frontend UI enhancements. The most notable changes include adding new routes for design management, enhancing backend data handling, and updating the frontend's footer and header components for improved styling and functionality.

### Backend Changes

#### New API Endpoints:
* Added a `GET /designs` endpoint to retrieve paginated user designs and a `DELETE /design` endpoint for deleting designs. Both endpoints include user authentication and authorization checks. (`app.py`, [[1]](diffhunk://#diff-568470d013cd12e4f388206520da39ab9a4e4c3c6b95846cbc281abc1ba3c959R146-R155) [[2]](diffhunk://#diff-568470d013cd12e4f388206520da39ab9a4e4c3c6b95846cbc281abc1ba3c959R187-R194)

#### Backend Logic Enhancements:
* Added a `get_user_designs` method in the `Design` class to fetch paginated designs with user restrictions and admin overrides. (`controller/design.py`, [controller/design.pyR95-R115](diffhunk://#diff-34f7147910c6bdb62f8990f80a05a73b98a0259c58291b6ad051932ac86bb089R95-R115))
* Introduced `is_design_scheduled` and `is_design_in_queue` methods in `QueueItemDAO` to check the scheduling and queuing status of designs. (`model/queue_item.py`, [model/queue_item.pyR225-R247](diffhunk://#diff-f4afe307ef404f87464446569f8f128c9bf6fc622846eab4fbae4b4820119bafR225-R247))
* Enhanced the `delete_design` method to handle cases where no design ID is provided or when no rows are deleted. (`controller/design.py`, [controller/design.pyR214-R216](diffhunk://#diff-34f7147910c6bdb62f8990f80a05a73b98a0259c58291b6ad051932ac86bb089R214-R216))

### Frontend Changes

#### Footer and Header Updates:
* Redesigned the `Footer` component with a more modern layout, including sponsor social links and improved styling using Tailwind CSS. (`view/src/components/Footer.jsx`, [view/src/components/Footer.jsxL2-R39](diffhunk://#diff-2b7943d99899a12f7df32710cd68e5708effa66c73e19014a1203dd9a3e80183L2-R39))
* Simplified the `Header` component by removing redundant code and improving the navigation tracking logic. (`view/src/components/Header.jsx`, [[1]](diffhunk://#diff-b9cfe8a5a50dbf4266d7ca856723d43f4ba1ad9a155a760b1b4ab0dff1b613ddL2-R5) [[2]](diffhunk://#diff-b9cfe8a5a50dbf4266d7ca856723d43f4ba1ad9a155a760b1b4ab0dff1b613ddL17-R19)

#### General UI Improvements:
* Removed unused CSS rules for content area height and overflow in `App.css`. (`view/src/App.css`, [view/src/App.cssL11-L17](diffhunk://#diff-21152c98198a4af2d38781e93fbe432563024b22371c16678f7004f8006287f4L11-L17))

These changes aim to enhance both the backend's functionality and the frontend's user experience, ensuring better performance and usability across the application.

edimarvalentin commented on model/design.py:
```diff
@@ -21,6 +21,21 @@ def get_design_by_id(self, design_id: int):
+            columns = [col[0] for col in cursor.description]
+            rows = cursor.fetchall()
+            result = [dict(zip(columns, row)) for row in rows]
+            return result
```
ok lol

edimarvalentin commented:
@PinkSylvie In the preview, the buttons show with the wrong styling. That's because that is my most recent screenshot, but the code actually reflects the changes you suggested.

edimarvalentin commented:
> @PinkSylvie In the preview, the buttons show with the wrong styling. That's because that is my most recent screenshot, but the code actually reflects the changes you suggested.

![WhatsApp Image 2025-04-27 at 15 48 17_6e982fb7](https://github.com/user-attachments/assets/c7ebbd30-6034-4616-b12c-9ed54102d8aa)

PinkSylvie commented:
![image](https://github.com/user-attachments/assets/f82df123-da88-4e8b-9b05-23f33d4b2d20)
Looks and works great! Only detail is the hover color of the buttons, it'd be nice for the background to be black instead similar to how it is on the other pages buttons 
![image](https://github.com/user-attachments/assets/7c14db98-f030-4bc5-9a9b-eaf81bbb32af)

edimarvalentin commented:
@PinkSylvie  Buttons now go black on hover. The delete button stays white.
![image](https://github.com/user-attachments/assets/c5bbeb08-f260-40f9-8f0d-f46ff18ac8d3)

PinkSylvie reviewed (approved):
Looks great!

irsaris reviewed (changes requested):
![image](https://github.com/user-attachments/assets/2102fd30-2442-4fb4-ad22-039badd41a34)
Not sure what happened to the header and footer, make sure that they are compatible across all screens like before.

![image](https://github.com/user-attachments/assets/9cb84084-4578-4000-96a9-f76bb89dec9f)
A bit confused as to where the other buttons in relation to the homepage went. Also fix the proportions/look, should be the same flexibility issue as before.

![image](https://github.com/user-attachments/assets/05da270e-9193-4d72-bfb8-ddf49cad2dca)
You didn't add the button functionality for the three buttons, also if an image is in an active queue the image cannot be edited, so button must be removed.
Also not sure what the 'status' is used for since, if its in the queue or not or even dispalyed its value remains unchanged.

My recommendation, leave the footer and header as it was and don't mess with the styling that already was there.

edimarvalentin commented:
@irsaris Thank you for the feedback! 

## In response to your request:

1.  Mouse now changes to the "pointing" state when hovering over the buttons. 
2. The edit button is bound to the `handleEdit` function within the component. From there, you have to route to the create view, but instead of going to the default view, add the logic to handle going to that view to edit a specified image. You have to make the component mount with a design passed to it. 
3. If a design is scheduled, a user can now only edit the schedule or delete it:
![Screenshot 2025-04-30 at 16-53-16 Vite React](https://github.com/user-attachments/assets/418591f4-f30f-4041-92b5-d145c19400ba)
4. I was confused about this one and I think this attribute didn't find  any real use in the  project timeline. It was removed from view. Design Id also removed since it has no value to the user. 
5. The built-in modal is now in use. It was updated to handle deletes and shows alerts (errors) to the user:
![Screenshot 2025-04-30 at 16-57-34 Vite React](https://github.com/user-attachments/assets/da1d33ef-c7fd-49dd-b08e-44f253674364)

## Some additional comments:

- I did some light styling changes to be consistent with the other views as much as my ability allows. 
- If a design has been set to "not approved", the user can only edit or delete it. 
- I fixed the overflowing text issue by truncating long design titles. 
- I did not find any serious issues with the new header and footer. I had to change the footer due to an issue where I saw that it wasn't pushing to the bottom properly with the main content, overlapping at times.  Styling is one of the easier and more flexible parts of the project, so it's expected to change as we go. If there's any dissatisfaction with the styling, the entire front-end can be restyled into a clean, consistent look within a day.
- If time allows, I'll add numbering to the pagination buttons as @PinkSylvie does in her views. 
- The buggiest part to work with was the delete function. When an image is deleted, it is removed from the gallery. Is it bug-free now to the best of my knowledge, but if you happen to find a weird logical bug when deleting and paginating around, please let me know. 
- Maybe the modal delete button should be red too?

irsaris commented:
![image](https://github.com/user-attachments/assets/772a7742-bf5b-4d89-953a-5348bc0ec895)
Styling for header and footer fixed, could you check the components of the View Saved Images page? It looks like they stick to the top.

edimarvalentin commented:
## Here are the new changes:

- Updated the labels to provide better context of design status.
- I updated the layout to better organize the content. I tested with different screen widths. Works well with anything but mobile, since the lack of scrolling makes vertical scaling a bit tricky. However, I was able to set scrolling for the gallery, allowing me to make more use of the width, while keeping the same height. 
![Screenshot 2025-04-30 at 22-45-56 Vite React](https://github.com/user-attachments/assets/f3f7f6d6-7bc3-42f7-a625-5431d77563ef)

@irsaris If if still showing up incomplete, like missing margins and wrong sizing, check your node_modules and make sure Tailwind is there as specified in the package.json. I don't think it should be a problem since part of the styles are loading for you, so that means it is installed, but just to be sure.

@PinkSylvie Feel free to test the view and confirm if there's a problem with the styling. There's a present problem I can't identify and I can't replicate Irsa's view as I adjust height and width. The elements scale properly for me.

### Thank you!

PinkSylvie commented:
![image](https://github.com/user-attachments/assets/5589be00-91db-45d6-a884-2245047d6155)
It works well till it doesn't. I'd put the scroll instead of on the images container on the container that has both the images and the image info

edimarvalentin commented:
@PinkSylvie I'll see if I can get the mobile viewport working.

edimarvalentin commented:
@PinkSylvie I fixed it for mobile viewports and made the entire component capable of scrolling. 
![Screenshot 2025-05-01 at 10-54-25 Vite React](https://github.com/user-attachments/assets/417e4599-5705-4527-b8cf-e7bed9bc04c4)

edimarvalentin commented:
> It works better but a little bit of concern on how the image data shows when on mobile ![{0C1D8378-1D12-4D5D-B8DB-E26EE837DC6B}](https://private-user-images.githubusercontent.com/93553908/439643321-319a23fc-7467-4d0f-9e49-a5cf164d7f2a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDYxMTQ3MzYsIm5iZiI6MTc0NjExNDQzNiwicGF0aCI6Ii85MzU1MzkwOC80Mzk2NDMzMjEtMzE5YTIzZmMtNzQ2Ny00ZDBmLTllNDktYTVjZjE2NGQ3ZjJhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTAxVDE1NDcxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFmNTk5YWRhMDk3NTkyYzI1Yjk3ZDRiZWRjMmYwNmM3NzA5MDUxOWI5OGY5ZTkzOTgwYzk3NDc0YTJmOGM0ODImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.u5K3jKHnwg_Coep5RpOq24kDscjRZJ9jZt8EWZVQvb4)

@PinkSylvie Yeah, that is the awkward spot between desktop and mobile. I would have to adjust the constraints. I'll tend to that after I finish the board executable.

edimarvalentin commented:
> > It works better but a little bit of concern on how the image data shows when on mobile ![{0C1D8378-1D12-4D5D-B8DB-E26EE837DC6B}](https://private-user-images.githubusercontent.com/93553908/439643321-319a23fc-7467-4d0f-9e49-a5cf164d7f2a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDYxMTQ3MzYsIm5iZiI6MTc0NjExNDQzNiwicGF0aCI6Ii85MzU1MzkwOC80Mzk2NDMzMjEtMzE5YTIzZmMtNzQ2Ny00ZDBmLTllNDktYTVjZjE2NGQ3ZjJhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTAxVDE1NDcxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFmNTk5YWRhMDk3NTkyYzI1Yjk3ZDRiZWRjMmYwNmM3NzA5MDUxOWI5OGY5ZTkzOTgwYzk3NDc0YTJmOGM0ODImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.u5K3jKHnwg_Coep5RpOq24kDscjRZJ9jZt8EWZVQvb4)
> 
> @PinkSylvie Yeah, that is the awkward spot between desktop and mobile. I would have to adjust the constraints. I'll tend to that after I finish the board executable.

@PinkSylvie Also, the *id* and *status*  attributes are showing on the card, and that's the header and footers I had to undo from the PR. Could you double check that 3641fe8a9d40f2b0e777aaa2f347e58c088153cc is the head of your branch?

irsaris commented:
Hello, tested it and made sure to have all the dependencies were installed. It still looks the same to me, try following @PinkSylvie 's structure of making a component and have it move. Also, labels haven't been done yet it seems, its green for 'In Queue' and blue for 'Scheduled'.
![image](https://github.com/user-attachments/assets/0fb0246a-98c2-4240-8e2c-1e4e5410e13b)

irsaris commented:
![image](https://github.com/user-attachments/assets/e79eda77-451e-45eb-9237-c1678200dbd9)
Looks good to me! @PinkSylvie thank you for helping out.

Only thing missing is the labelling differentiation between unscheduled and scheduled items, I'll take care of that in my branch.
